File: src/main/java/me/arasple/mc/trmenu/action/acts/ActionSetArgs.java
package me.arasple.mc.trmenu.action.acts;

import me.arasple.mc.trmenu.action.base.AbstractAction;
import me.arasple.mc.trmenu.data.ArgsCache;
import org.bukkit.entity.Player;

/**
 * @author Arasple
 * @date 2020/1/18 20:48
 */
public class ActionSetArgs extends AbstractAction {

    @Override
    public String getName() {
        return "set(-)?args";
    }

    @Override
    public void onExecute(Player player) {
        ArgsCache.updateArgs(player.getUniqueId(), getContent(player).split(" "));
    }

}

File: src/main/kotlin/me/arasple/mc/trmenu/modules/action/impl/menu/ActionSetArgs.kt
package me.arasple.mc.trmenu.modules.action.impl.menu

import me.arasple.mc.trmenu.data.MetaPlayer.setArguments
import me.arasple.mc.trmenu.modules.action.base.Action
import me.arasple.mc.trmenu.utils.Msger
import org.bukkit.entity.Player

/**
 * @author Arasple
 * @date 2020/3/8 21:33
 */
class ActionSetArgs : Action("set(-)?arg(ument)?(s)?") {

    override fun onExecute(player: Player) {
        player.setArguments(
            getContentSplited(player, " ")
                .map { Msger.replace(player, replaceWithSpaces(it)) }
                .toTypedArray()
        )
    }

}